programs/protocol/src/data/carrier.rs

```diff
@@ -0,0 +1,20 @@
+use anchor_lang::prelude::{
+    borsh::{BorshDeserialize, BorshSerialize},
+    *,
+};
+
+use crate::GeoLocation;
+
+#[zero_copy]
+#[derive(Debug, Default, PartialEq, BorshSerialize, BorshDeserialize)]
+pub struct Availability {
+    pub time: u64, // when 0, carrier is not available
+    pub location: GeoLocation,
+}
+
+#[account(zero_copy)]
+#[derive(Debug, Default, PartialEq, BorshSerialize, BorshDeserialize)]
+pub struct Carrier {
+    pub authority: Pubkey,
+    pub availability: Availability,
```

Comment:
just one?

Reply:
All the future availability would depend on the next shipment

We can add longer-term planning later, but for now I think the map with available time is more than enough